src/components/tableV2/hooks/useInfiniteScroll.js

```diff
@@ -0,0 +1,139 @@
+import { useEffect, useMemo, useState, useRef } from "react"
+
+//COMMENTS HERE ARE ONLY FOR TESTING REASONS THE ARE GOING TO BE REMOVED AFTER WE SEE THAT THE INFINITY WORKS AS EXPECTED
+
+// const fakeData = () => [
+//   {
+//     nodes: "node51",
+//     alerts: 15,
+//     user: "mitsos",
+//     disabled: true,
+//     status: "stale",
+//     untouchable: "true",
+//   },
+//   {
+//     nodes: "node52",
+//     alerts: 15,
+//     user: "mitsos",
+//     disabled: true,
+//     status: "stale",
+//     untouchable: "true",
+//   },
+//   {
+//     nodes: "node53",
+//     alerts: 15,
+//     user: "mitsos",
+//     disabled: true,
+//     status: "stale",
+//     untouchable: "true",
+//   },
+//   {
+//     nodes: "node54",
+//     alerts: 15,
+//     user: "mitsos",
+//     disabled: true,
+//     status: "stale",
+//     untouchable: "true",
+//   },
+//   {
+//     nodes: "node55",
+//     alerts: 15,
+//     user: "mitsos",
+//     disabled: true,
+//     status: "stale",
+//     untouchable: "true",
+//   },
+//   {
+//     nodes: "node56",
+//     alerts: 15,
+//     user: "mitsos",
+//     disabled: true,
+//     status: "stale",
+//     untouchable: "true",
+//   },
+// ]
+
+// function loadMoreItems(lastData) {
+//   console.log(lastData)
+//   return new Promise(resolve => {
+//     setTimeout(() => {
+//       resolve({
+//         data: fakeData(),
+//       })
+//     }, 1000)
+//   })
+// }
+
+const useScrollListener = (target, handler) => {
+  const listenerRef = useRef(handler)
+  listenerRef.current = handler
+
+  const listener = () => listenerRef.current()
```

Comment:
nit: This can be created inside the useEffect, to avoid recreating in each rerender.